### PR TITLE
BL-7019 new quiz pages are numbered and omitted from epubs

### PR DESCRIPTION
**Pull request by JohnThomson**
This change is [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/3280)

### CODE REVIEW

**Review by andrew-polk — CHANGES_REQUESTED** [review]
> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/3280)* status: all files reviewed, 2 unresolved discussions (waiting on @JohnThomson)

___
*[src/BloomExe/Publish/Epub/EpubMaker.cs, line 978 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/3280#-LiOl6K_4jjPYyg7BP6P:-LiOl6K_4jjPYyg7BP6Q:b-dyfpgc) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/675e81ac65c9deaa02e3286e9464fcce9dff350a/src/BloomExe/Publish/Epub/EpubMaker.cs#L978)):*
> ```Smalltalk
> 				{
> 					_reportedOmittedInteractivePages = true;
> 					progress.Message("OmittedInteractive", "Interactive pages (like comprehension questions) are not supported in epubs, and will be left out", MessageKind.Warning);
> ```

I'm unsure about this message. Currently we also remove "Translation Instructions". In the future, there may be other types altogether.
In other words, the message seems too specific.
 
One possibility (not saying I love it):
Keep a list of all page labels we will remove, then print one message at the end like,
The following pages were removed as they are not supported in epubs:
Translation Instructions
Quiz Page

(only listing each page label once, of course)
___
*[src/BloomTests/Publish/ExportEpubTests.cs, line 522 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/3280#-LiOkEgIDTPniXa1II6K:-LiOkEgIDTPniXa1II6L:b3rv0sf) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/675e81ac65c9deaa02e3286e9464fcce9dff350a/src/BloomTests/Publish/ExportEpubTests.cs#L522)):*
> ```Smalltalk
> 
> 			var page2entry = _epub.GetEntry(Path.GetDirectoryName(_manifestFile) + "/" + "2.xhtml");
> 			Assert.That(page2entry, Is.Null, "nonprinting page should be omitted");
> ```

I think we want another assert here showing that the interactive page was removed. And the comment should indicate why we want to do so.

**Review by hatton — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/3280)* status: all files reviewed, 2 unresolved discussions (waiting on @andrew-polk and @JohnThomson)

___
*[src/BloomExe/Publish/Epub/EpubMaker.cs, line 978 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/3280#-LiOl6K_4jjPYyg7BP6P:-LiOmXt655R_Cwx_YGH4:b-nsxga4) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/675e81ac65c9deaa02e3286e9464fcce9dff350a/src/BloomExe/Publish/Epub/EpubMaker.cs#L978)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

I'm unsure about this message. Currently we also remove "Translation Instructions". In the future, there may be other types altogether.
In other words, the message seems too specific.
 
One possibility (not saying I love it):
Keep a list of all page labels we will remove, then print one message at the end like,
The following pages were removed as they are not supported in epubs:
Translation Instructions
Quiz Page

(only listing each page label once, of course)
</blockquote></details>

I like that

**Review by JohnThomson — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/3280)* status: 1 of 6 files reviewed, 2 unresolved discussions (waiting on @andrew-polk)

___
*[src/BloomExe/Publish/Epub/EpubMaker.cs, line 978 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/3280#-LiOl6K_4jjPYyg7BP6P:-LiPGApK6owSrNjkR3jW:b-uwwdb0) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/675e81ac65c9deaa02e3286e9464fcce9dff350a/src/BloomExe/Publish/Epub/EpubMaker.cs#L978)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

I like that
</blockquote></details>

Done. (But should we do it for BR, too? So far the only thing we omit there is translation instructions, AFAIK, and that seems clear enough in the description not to need a warning.)
___
*[src/BloomTests/Publish/ExportEpubTests.cs, line 522 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/3280#-LiOkEgIDTPniXa1II6K:-LiP9WIx6xVqo62pWWPf:b-x2xskm) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/675e81ac65c9deaa02e3286e9464fcce9dff350a/src/BloomTests/Publish/ExportEpubTests.cs#L522)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

I think we want another assert here showing that the interactive page was removed. And the comment should indicate why we want to do so.
</blockquote></details>

I improved the message; and I did check that the test fails without the new code. I think the comment about why we don't want such pages more belongs in the main code (where I have one).

**Review by andrew-polk — CHANGES_REQUESTED** [review]
> Reviewed 5 of 5 files at r2.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/3280)* status: all files reviewed, 2 unresolved discussions (waiting on @JohnThomson)

___
*[DistFiles/localization/en/Bloom.xlf, line 3243 at r2](https://reviewable.io/reviews/bloombooks/bloomdesktop/3280#-LiQNxyM9zowKOHMZskq:-LiQNxyM9zowKOHMZskr:b8t6t1v) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/58366879e90793d05abfcbebde4313889945d0da/DistFiles/localization/en/Bloom.xlf#L3243)):*
> ```XML
>       </trans-unit>
>       <trans-unit id="PublishTab.Epub.OmittedPages">
>         <source xml:lang="en">Interactive pages (like comprehension questions) are not supported in epubs, and will be left out</source>
> ```

The message didn't get changed to match the one in the code.
___
*[src/BloomExe/Publish/Epub/EpubMaker.cs, line 978 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/3280#-LiOl6K_4jjPYyg7BP6P:-LiQNXZ5BCD16bVlRbcn:b559nl3) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/675e81ac65c9deaa02e3286e9464fcce9dff350a/src/BloomExe/Publish/Epub/EpubMaker.cs#L978)):*
<details><summary><i>Previously, JohnThomson (John Thomson) wrote…</i></summary><blockquote>

Done. (But should we do it for BR, too? So far the only thing we omit there is translation instructions, AFAIK, and that seems clear enough in the description not to need a warning.)
</blockquote></details>

I agree it doesn't seem necessary at this point.
___
*[src/BloomExe/Publish/Epub/EpubMaker.cs, line 317 at r2](https://reviewable.io/reviews/bloombooks/bloomdesktop/3280#-LiQO7O7CroTH7cQzdAw:-LiQO7O7CroTH7cQzdAx:bi2tp9w) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/58366879e90793d05abfcbebde4313889945d0da/src/BloomExe/Publish/Epub/EpubMaker.cs#L317)):*
> ```Smalltalk
> 			{
> 				progress.Message("OmittedPages",
> 					"The following pages were removed as they are not supported in epubs:",
> ```

Sorry; I realized after I wrote the initial suggestion that it should be 
`The following pages were removed because they are not supported in ePUBs:`

Note as => because and epubs => ePUBs.
___
*[src/BloomTests/Publish/ExportEpubTests.cs, line 522 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/3280#-LiOkEgIDTPniXa1II6K:-LiQNoEs8oWOLkBfgqYF:bfq2u6) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/675e81ac65c9deaa02e3286e9464fcce9dff350a/src/BloomTests/Publish/ExportEpubTests.cs#L522)):*
<details><summary><i>Previously, JohnThomson (John Thomson) wrote…</i></summary><blockquote>

I improved the message; and I did check that the test fails without the new code. I think the comment about why we don't want such pages more belongs in the main code (where I have one).
</blockquote></details>

Right, I meant the message. And I wasn't thinking clearly regarding needing a new Assert. Thanks.

**Review by JohnThomson — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/3280)* status: all files reviewed, 2 unresolved discussions (waiting on @andrew-polk)

___
*[DistFiles/localization/en/Bloom.xlf, line 3243 at r2](https://reviewable.io/reviews/bloombooks/bloomdesktop/3280#-LiQNxyM9zowKOHMZskq:-LiTNCKlDTTKa9usTLug:b-896fix) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/58366879e90793d05abfcbebde4313889945d0da/DistFiles/localization/en/Bloom.xlf#L3243)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

The message didn't get changed to match the one in the code.
</blockquote></details>

Done.
___
*[src/BloomExe/Publish/Epub/EpubMaker.cs, line 317 at r2](https://reviewable.io/reviews/bloombooks/bloomdesktop/3280#-LiQO7O7CroTH7cQzdAw:-LiTNBhR5ATvpKjYOhDi:b-896fix) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/58366879e90793d05abfcbebde4313889945d0da/src/BloomExe/Publish/Epub/EpubMaker.cs#L317)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Sorry; I realized after I wrote the initial suggestion that it should be 
`The following pages were removed because they are not supported in ePUBs:`

Note as => because and epubs => ePUBs.
</blockquote></details>

Done.

**Review by andrew-polk — APPROVED** [review]
> Reviewed 2 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/3280)* status: :shipit: complete! all files reviewed, all discussions resolved
